test/testrunner.py

```diff
@@ -259,6 +260,15 @@ def main():
     benchmark_mode = "benchmark" in test_type or is_benchmark_dev_context()
     specific_test_type = re.sub("benchmark-", "", test_type) if "benchmark" in test_type else test_type
     build_context = get_build_context()
+    # PR test executions can be distinguished by commit id. 
+    # But other pipelines triggered with the same commit id. 
+    # Adding codebuild_initiator to a pipeline name for distinguishing different build pipelines 
+    if not "PR" == build_context:
```

Comment:
There should be a function called `is_pr_context()` - however, are there specific contexts where this condition should apply? Should this also run in canary, for instance, or just MAINLINE and NIGHTLY?

Reply:
Additionally, I'm not sure I totally understand the reason for not considering the PR context. If we add the ability to run multiple types of jobs in parallel (i.e. graviton alongside regular cpu/gpu), then we'll run into the same problem again, right?

Reply:
PR jobs are triggered with different commit IDs. There won't be 2 PR jobs with the same commit ID. So we don't have to do it for PR pipelines

Reply:
We should keep it consistent. If we add the ability to run graviton and cpu/gpu from the same commit, for example, we'll have to retroactively track this down and remove this condition.